tests/dir_helpers.py

```diff
@@ -285,3 +293,21 @@ def erepo_dir(tmp_path_factory, monkeypatch):
         path.dvc.close()
 
     return path
+
+
+@pytest.fixture
+def git_dir(tmp_path_factory, monkeypatch):
```

Comment:
I would suggest renaming it so that we know its "external git dir".
Also, if we are already initializing git repository here, how about refactoring `erepo_dir` so that it uses this fixture? If you consider my point valid, `path` generation should have logic basing on `request.fixturenames` (e.g. https://github.com/iterative/dvc/blob/482473aa69fd97ebde8bdd2ae23b0c703158f765/tests/dir_helpers.py#L223)

Reply:
I don't think it's very helpful to make this one react to other things. For example, the test `test_update_git_tracked` which I've added, needs DVC on the tmp_dir, but not on the external git dir. This would make this fixture useless for that test, since it would have DVC anyway.

We could refactor erepo_dir to use this fixture, but that would mean two commits, and the tests are pretty slow as it is right now.